mapiproxy/servers/default/asyncemsmdb/dcesrv_asyncemsmdb.c

```diff
@@ -0,0 +1,1230 @@
+
+	retval = mapistore_properties_get_properties(p->mstore_ctx, contextID, context_object, mem_ctx, properties->cValues,
+						     properties->aulPropTag, prop_data);
+	MAPISTORE_RETVAL_IF(retval, retval, prop_data);
+
+	for (i = 0; i < properties->cValues; i++) {
+		if (prop_data[i].error) {
+				if (properties->aulPropTag[i] == PidTagFolderFlags) {
+					prop_data[i].data = talloc_zero(prop_data, uint32_t);
+					*((uint32_t *)prop_data[i].data) = FolderFlags_IPM|FolderFlags_Normal;
+					data_pointers[i] = prop_data[i].data;
+					(void) talloc_reference(data_pointers, prop_data[i].data);
+					retvals[i] = MAPI_E_SUCCESS;
+				} else {
+					retvals[i] = mapistore_error_to_mapi(prop_data[i].error);
+				}
+		} else {
+			if (prop_data[i].data == NULL) {
+				if (properties->aulPropTag[i] == PidTagFolderFlags) {
+					prop_data[i].data = talloc_zero(prop_data, uint32_t);
+					*(uint32_t *)prop_data[i].data = FolderFlags_IPM|FolderFlags_Normal;
+					data_pointers[i] = prop_data[i].data;
+					(void) talloc_reference(data_pointers, prop_data[i].data);
+					retvals[i] = MAPI_E_SUCCESS;
+				} else {
+					retvals[i] = MAPI_E_NOT_FOUND;
+				}
+			} else {
+				if (properties->aulPropTag[i] == PidTagParentFolderId) {
+					if (*(uint64_t *)prop_data[i].data == 0x0) {
+						uint64_t	parent_fid;
+
+						retval = get_mapistore_sogo_PidTagParentFolderId(mem_ctx, p, uri, &parent_fid);
+						if (retval == MAPISTORE_SUCCESS) {
+							*(uint64_t *)(prop_data[i].data) = parent_fid;
+							retvals[i] = MAPI_E_SUCCESS;
+						}
+					}
+				} else if (properties->aulPropTag[i] == PidTagFolderFlags) {
+					prop_data[i].data = talloc_zero(prop_data, uint32_t);
+					*(uint32_t *)prop_data[i].data = FolderFlags_IPM|FolderFlags_Normal;
+					data_pointers[i] = prop_data[i].data;
+					retvals[i] = MAPI_E_SUCCESS;
+				}
+				data_pointers[i] = prop_data[i].data;
+				(void) talloc_reference(data_pointers, prop_data[i].data);
+			}
+		}
+	}
+
+	/* Delete mapistore context */
+	retval = mapistore_del_context(p->mstore_ctx, contextID);
+	MAPISTORE_RETVAL_IF(retval, retval, prop_data);
+
+	talloc_free(prop_data);
+	return MAPISTORE_SUCCESS;
+}
+
+
+/**
+   \details Retrieve properties from openchangedb system folder
+
+   \param mem_ctx pointer to the memory context to use to allocate returned data
+   \param p pointer to the private asyncemsmdb data
+   \param folderId the ID of the folder from which data are to be retrieved
+   \param properties pointer to the array of MAPI properties to retrieve
+   \param data_pointers pointer on pointer to the data to return
+   \param retvals pointers to the return value of each data_pointers entry to return
+
+   \return MAPI_E_SUCCESS on success, otherwise MAPI error
+ */
+static enum MAPISTATUS get_properties_systemspecialfolder(TALLOC_CTX *mem_ctx, struct asyncemsmdb_private_data *p,
+							  uint64_t folderId, struct SPropTagArray *properties,
+							  void **data_pointers, enum MAPISTATUS *retvals)
+{
```

Comment:
Why don't you use `get_properties_openchangedb` likewise `mapistore` is used above, instead of subfolder?